File: code/1207.unique-number-of-occurrences.php
<?php
/*
 * @lc app=leetcode id=1207 lang=php
 *
 * [1207] Unique Number of Occurrences
 */

// @lc code=start
class Solution {

    /**
     * @param Integer[] $arr
     * @return Boolean
     */
    function uniqueOccurrences($arr) {
        // my solution
        $countMap = array_count_values($arr);
        $processed = [];
        foreach ($countMap as $count) {
            if (in_array($count, $processed)) {
                return false;
            }
            $processed[] = $count;
        }
        return true;

        // e.g.
        // $map = array_count_values($arr);
        // return count(array_unique($map)) === count($map);
    }
}
// @lc code=end

File: code/1207.unique-number-of-occurrences.go
package code

/*
 * @lc app=leetcode id=1207 lang=golang
 *
 * [1207] Unique Number of Occurrences
 */

// @lc code=start
func uniqueOccurrences(arr []int) bool {
	table := make(map[int]int)
	for _, v := range arr {
		table[v] += 1
	}

	check := make(map[int]bool)
	for _, v := range table {
		if !check[v] {
			check[v] = true
		} else {
			return false
		}
	}

	return true
}

// @lc code=end
